compositor_pipeline/src/pipeline/encoder/ffmpeg_h264.rs

```diff
@@ -0,0 +1,295 @@
+        encoder.set_format(Pixel::YUV420P);
+        encoder.set_width(options.resolution.width as u32);
+        encoder.set_height(options.resolution.height as u32);
+
+        let mut encoder = encoder.open_as_with(
+            codec,
+            // TODO: audit settings bellow
+            // Those values are copied from somewhere, they have to be set because libx264
+            // is throwing an error if it detects default ffmpeg settings.
+            Dictionary::from_iter([
+                ("preset", options.preset.to_str()),
+                // Quality-based VBR (0-51)
+                ("crf", "23"),
+                // Override ffmpeg defaults from https://github.com/mirror/x264/blob/eaa68fad9e5d201d42fde51665f2d137ae96baf0/encoder/encoder.c#L674
+                // QP curve compression - libx264 defaults to 0.6 (in case of tune=grain to 0.8)
+                ("qcomp", "0.6"),
+                //  Maximum motion vector search range - libx264 defaults to 16 (in case of placebo
+                //  or veryslow preset to 24)
+                ("me_range", "16"),
+                // Max QP step - libx264 defaults to 4
+                ("qdiff", "4"),
+                // Min QP - libx264 defaults to 0
+                ("qmin", "0"),
+                // Max QP - libx264 defaults to QP_MAX = 69
+                ("qmax", "69"),
+                //  Maximum GOP (Group of Pictures) size - libx264 defaults to 250
+                ("g", "250"),
+                // QP factor between I and P frames - libx264 defaults to 1.4 (in case of tune=grain to 1.1)
+                ("i_qfactor", "1.4"),
+                // QP factor between P and B frames - libx264 defaults to 1.4 (in case of tune=grain to 1.1)
+                ("f_pb_factor", "1.3"),
+                // A comma-separated list of partitions to consider. Possible values: p8x8, p4x4, b8x8, i8x8, i4x4, none, all
+                ("partitions", options.preset.default_partitions()),
+                // Subpixel motion estimation and mode decision (decision quality: 1=fast, 11=best)
+                ("subq", options.preset.default_subq_mode()),
+            ]),
+        )?;
+
+        result_sender.send(Ok(())).unwrap();
+
+        loop {
+            let frame = match frame_receiver.recv() {
+                Ok(Message::Frame(f)) => f,
+                Ok(Message::Stop) => break,
+                Err(_) => break,
+            };
+
+            if frame_receiver.len() > 20 {
+                warn!("Dropping frame: render queue is too long.",);
+                continue;
+            }
+
+            let mut av_frame = frame::Video::new(
+                Pixel::YUV420P,
+                options.resolution.width as u32,
+                options.resolution.height as u32,
+            );
+
+            frame_into_av(frame, &mut av_frame);
+
+            if let Err(e) = encoder.send_frame(&av_frame) {
+                error!("Encoder error: {e}.")
+            }
```

Comment:
Inside frame_into_av and when calling encoder.send_frame you are login in error, but do not change any control flow in response. The resulting behavior is not obvious.

If `frame_into_av` fails you will send empty frame (either green or random if initial frame does not zero its memory)

If `send_frame` fails you will log error and receive_packet will not return anything, so behvior is correct, but I think it would be cleaner to handle those errors explicitlly